### PR TITLE
refactor(plugingen): make this script more extensible by introducing generation phases

### PR DESCRIPTION
**Pull request by moldy530**
# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

### Focus of this PR:
This PR focuses on improving the plugin generation process and updating dependencies.

### Detailed summary:
- Updated `tsconfig.build.json` to include additional directories for compilation.
- Updated `.eslintignore` to include additional directories for linting.
- Updated `wagmi.config.ts` to import `plugingen` from a different file location.
- Added new files and functions for plugin generation.
- Updated `package.json` to include additional directories and files.
- Updated plugin files to import and use new dependencies.
- Updated plugin files to use new functions and generate new code.
- Updated plugin files to import and use new types.
- Updated plugin files to import and use new interfaces.
- Updated plugin files to import and use new constants.
- Updated plugin files to import and use new utility functions.

> The following files were skipped due to too many changes: `packages/accounts/plugingen/phases/plugin-generator/provider-method-gen/index.ts`, `packages/accounts/plugingen/phases/plugin-generator/account-method-gen.ts`, `packages/accounts/plugingen/phases/plugin-generator/provider-method-gen/install-method-gen.ts`, `packages/accounts/plugingen/index.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: packages/accounts/plugingen/index.ts [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        }
+
+        types.set(typeName, {
+          definition: typeDef.replace(";", ""),
+          isPublic: isPublic ?? false,
+        });
+      };
+
+      if (contracts.length !== 1) {
+        console.warn(
```
Comment (denniswon) [inline packages/accounts/plugingen/index.ts:68]: why not just throw here?

Comment (moldy530) [reply packages/accounts/plugingen/index.ts:68]: yea good question.... I will throw here

File: packages/accounts/plugingen/phases/plugin-generator/provider-method-gen/install-method-gen.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      return installPlugin_(provider, {
+        pluginAddress,
+        pluginInitData: encodeAbiParameters(${JSON.stringify(
+          initArgs
+        )}, params.args),
+        dependencies,
+      }, overrides);
+    }
+  `);
+  }
```
Comment (denniswon) [inline packages/accounts/plugingen/phases/plugin-generator/provider-method-gen/install-method-gen.ts:71]:
```suggestion
    input.content.push(dedent`
      install${contract.name}: (params: Install${
        contract.name
      }Params, overrides?: UserOperationOverrides) => {
        const chain = provider.rpcClient.chain;
        const dependencies = params.dependencyOverrides ?? [${dependencies.join(
          ",\n\n"
        )}];
        const pluginAddress = params.pluginAddress ?? ${
          contract.name
        }_.meta.addresses[chain.id] as Address | undefined;
  
        if (!pluginAddress) {
          throw new Error("missing ${
            contract.name
          } address for chain " + chain.name);
        }
        
        return installPlugin_(provider, {
          pluginAddress,
          pluginInitData: encodeAbiParameters(${JSON.stringify(
            initArgs
          )}, params.args),
          dependencies,
        }, overrides);
      }
    `);
  }
```

**Review by denniswon — APPROVED** [review]

File: packages/accounts/plugingen/index.ts [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+      const addType: PhaseInput["addType"] = (typeName, typeDef, isPublic) => {
+        if (types.has(typeName)) {
+          throw new Error(`Type ${typeName} already exists`);
+        }
+
+        types.set(typeName, {
+          definition: typeDef.replace(";", ""),
+          isPublic: isPublic ?? false,
+        });
+      };
```
[the remark this answers is not in the document]
Comment (moldy530) [reply packages/accounts/plugingen/index.ts:65]: yea good call

File: packages/accounts/plugingen/phases/plugin-generator/provider-method-gen/install-method-gen.ts (hunk 2 above)
Comment (moldy530) [reply packages/accounts/plugingen/phases/plugin-generator/provider-method-gen/install-method-gen.ts:71]: oh yea... so prettier was autoformatting these

**Pull request merged**